### PR TITLE
Add more utility functions around modules and exports

### PR DESCRIPTION
**Pull request by hansl**
This allows to create a moderately simple example (see `tests/gcd.rs`) that can be used as a tutorial.

There are some typing issues, notably the `name` argument in `get_typed_fn`. Let's discuss.

### CODE REVIEW

File: core/engine/src/module/mod.rs [hunk 1]
```diff
@@ -566,6 +570,20 @@ impl Module {
         promise
     }
 
+    /// Utility function that execute [`load_link_evaluate`] and then run the job queue.
+    /// This combines two operations that are often done together into a simpler
+    /// version. If you need more control over the execution, use [`load_link_evaluate`]
+    /// and [`Context::run_jobs`] separately.
+    pub fn load_link_evaluate_and_run_jobs(&self, context: &mut Context) -> JsResult<()> {
```
Comment (jedel1043) [inline core/engine/src/module/mod.rs:577]: Maybe it's time to find a better name for the original function. `load_link_evaluate` sounds a bit spec-y and I think something like `run` would also be fine. That would also make the name of this function a lot shorter; `run_blocking` or something like that.

Comment (hansl) [reply core/engine/src/module/mod.rs:577]: `eval_blocking`? Either way I can make the change in this PR.

Comment (jedel1043) [reply core/engine/src/module/mod.rs:577]: Though about something. I think people would also want to await other promises, so maybe it should be better to add an `await_` method to `JsPromise`? That sidesteps the need for a new method for the module, and you would just call

`module.load_link_resolve(context).await_(context)?`

Comment (hansl) [reply core/engine/src/module/mod.rs]: PTAL

File: core/engine/src/object/builtins/jsfunction.rs [hunk 2]
```diff
@@ -1,11 +1,74 @@
...
 use std::ops::Deref;
 
+/// A trait for converting a tuple of Rust values into a vector of `JsValue`,
+/// to be used as arguments for a JavaScript function.
+pub trait TryIntoJsArguments {
+    /// Convert a tuple of Rust values into a vector of `JsValue`.
+    /// This is automatically implemented for tuples that implement
+    /// `TryIntoJsResult`.
+    fn into_js_args(self, cx: &mut Context) -> JsResult<Vec<JsValue>>;
+}
```
Comment (jedel1043) [inline core/engine/src/object/builtins/jsfunction.rs:17]: Praise: Loving this new API! Should help for strictness in many ways.

**Review by jedel1043 — APPROVED** [review]
Looks perfect to me!

File: core/engine/src/object/builtins/jsfunction.rs [hunk 3]
```diff
@@ -1,11 +1,74 @@
...
+/// but will not be verified at runtime (since JavaScript doesn't
+/// actually have strong typing).
+///
+/// To create this type, use the [`JsFunction::typed`] method.
+#[derive(Debug, Clone, Trace, Finalize)]
+pub struct TypedJsFunction<A: TryIntoJsArguments, R: TryFromJs> {
+    inner: JsFunction,
+    _args: PhantomData<A>,
+    _ret: PhantomData<R>,
+}
```
Comment (jedel1043) [inline core/engine/src/object/builtins/jsfunction.rs:48]: Loving this new API! It should help for strictness in many ways.

**Review by HalidOdat — APPROVED** [review]
Looks great! :)

**Pull request merged**